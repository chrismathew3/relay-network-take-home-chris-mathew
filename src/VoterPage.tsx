import React from "react";
import { Box } from "@chakra-ui/react";
import VoterPagePercentageSection from "./VoterPagePercentageSection";
import VoterPageTable from "./VoterPageTable";

const VoterPage: React.FC = () => {
  return (
    <Box id="VoterPage">
      <VoterPagePercentageSection/>
      <VoterPageTable />
    </Box>
  );
};

export default VoterPage;
