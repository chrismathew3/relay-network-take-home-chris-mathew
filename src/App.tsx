import VoterPage from './VoterPage'

function App() {

  return (
   <VoterPage/>
  )
}

export default App
