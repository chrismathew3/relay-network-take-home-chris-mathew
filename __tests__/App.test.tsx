import React from "react";
import renderer from "react-test-renderer";
import App from "../src/App";
import VoterPage from "../src/VoterPage";
const data = {
  rows: [
    {
      the_geom: null,
      the_geom_webmercator: null,
      ward: "Totals:",
      dem: 798647,
      rep: 116884,
      other_party: 119889,
      total: 1035420,
      white: 189630,
      black: 291084,
      hispanic: 69485,
      other_race: 33868,
      male: 337264,
      female: 417251,
      unknown_sex: 280905,
    },
  ],
  time: 0.003,
  fields: {
    the_geom: {
      type: "geometry",
      wkbtype: "Unknown",
      dims: 2,
      srid: 4326,
    },
    the_geom_webmercator: {
      type: "geometry",
      wkbtype: "Unknown",
      dims: 2,
      srid: 3857,
    },
    ward: {
      type: "string",
      pgtype: "text",
    },
    dem: {
      type: "number",
      pgtype: "int4",
    },
    rep: {
      type: "number",
      pgtype: "int4",
    },
    other_party: {
      type: "number",
      pgtype: "int4",
    },
    total: {
      type: "number",
      pgtype: "int4",
    },
    white: {
      type: "number",
      pgtype: "int4",
    },
    black: {
      type: "number",
      pgtype: "int4",
    },
    hispanic: {
      type: "number",
      pgtype: "int4",
    },
    other_race: {
      type: "number",
      pgtype: "int4",
    },
    male: {
      type: "number",
      pgtype: "int4",
    },
    female: {
      type: "number",
      pgtype: "int4",
    },
    unknown_sex: {
      type: "number",
      pgtype: "int4",
    },
  },
  total_rows: 67,
};

jest.mock("../src/hooks/useVoterStats", () => {
  return jest.fn(() => ({
    data: { data },
  }));
});

describe("<App/>", () => {
  test("<VoterPage/> should render", () => {
    const component = renderer.create(<App />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe("<VoterPage/>", () => {
  let component;
  let tree;
  beforeEach(() => {
    component = renderer.create(<VoterPage />);
    tree = component.toJSON();
  });

  afterEach(() => {
    // restore the spy created with spyOn
    jest.restoreAllMocks();
  });
  

  test("<VoterPage/> first tag should be be <div/> and should match snapshot", () => {
    expect(
      tree.type
    ).toEqual("div");
    expect(
      tree.props.id
    ).toEqual("VoterPage");
    expect(tree).toMatchSnapshot();
  });

  test("<VoterPage/> children should be <VoterPagePercentageSection/> and <VoterPageTable/> ", () => {
    expect(
      tree.children[0].type
    ).toEqual("div");
    expect(
      tree.children[0].props.id
    ).toEqual("VoterPagePercentageSection");
    expect(
      tree.children[1].type
    ).toEqual("div");
    expect(
      tree.children[1].props.id
    ).toEqual("VoterPageTable");
  });

  test("<VoterPagePercentageSection/> children should be <VoterPageSegmentDropDown/> and <h2/> ", () => {
    expect(
      tree.children[0].children[0].type
    ).toEqual("h2");
    expect(
      tree.children[0].children[1].type
    ).toEqual("h2");
    expect(
      tree.children[0].children[2].type
    ).toEqual("div");
    console.log(tree.children[0].children[2].children[0].props.id)
    expect(
      tree.children[0].children[2].children[0].props.id
    ).toEqual("VoterPageSegmentDropDown");
    expect(
      tree.children[0].children[2].children[0].children.length
     ).toEqual(12);
  });
});
