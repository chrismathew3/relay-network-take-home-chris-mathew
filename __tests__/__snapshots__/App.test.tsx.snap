// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App/> <VoterPage/> should render 1`] = `
<div
  className="css-0"
  id="VoterPage"
>
  <div
    className="css-9jay18"
    id="VoterPagePercentageSection"
  >
    <h2
      className="chakra-heading css-175t7kh"
    >
      Top Segment of All Voters: dem - 798647 - 77.13%
    </h2>
    <h2
      className="chakra-heading css-175t7kh"
    >
      Percentage of All Voters that are  - NaN - NaN%
    </h2>
    <div
      className="chakra-select__wrapper css-42jcko"
    >
      <select
        className="chakra-select css-1ik61og"
        id="VoterPageSegmentDropDown"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        value=""
      >
        <option
          value=""
        >
          Select Segment
        </option>
        <option
          value="dem"
        >
          dem
        </option>
        <option
          value="rep"
        >
          rep
        </option>
        <option
          value="other_party"
        >
          other_party
        </option>
        <option
          value="total"
        >
          total
        </option>
        <option
          value="white"
        >
          white
        </option>
        <option
          value="black"
        >
          black
        </option>
        <option
          value="hispanic"
        >
          hispanic
        </option>
        <option
          value="other_race"
        >
          other_race
        </option>
        <option
          value="male"
        >
          male
        </option>
        <option
          value="female"
        >
          female
        </option>
        <option
          value="unknown_sex"
        >
          unknown_sex
        </option>
      </select>
      <div
        className="chakra-select__icon-wrapper css-162mkon"
      >
        <svg
          aria-hidden={true}
          className="chakra-select__icon"
          focusable={false}
          role="presentation"
          style={
            {
              "color": "currentColor",
              "height": "1em",
              "width": "1em",
            }
          }
          viewBox="0 0 24 24"
        >
          <path
            d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
            fill="currentColor"
          />
        </svg>
      </div>
    </div>
  </div>
  <div
    className="chakra-table__container css-2l2gn2"
    id="VoterPageTable"
  >
    <table
      className="chakra-table css-0"
    >
      <caption
        className="css-18i7bby"
      >
        Philadelphia Qualified Voter Listing 2018
      </caption>
      <thead
        className="css-0"
      >
        <tr
          className="css-0"
        >
          <th
            className="css-0"
          >
            dem
          </th>
          <th
            className="css-0"
          >
            rep
          </th>
          <th
            className="css-0"
          >
            other_party
          </th>
          <th
            className="css-0"
          >
            total
          </th>
          <th
            className="css-0"
          >
            white
          </th>
          <th
            className="css-0"
          >
            black
          </th>
          <th
            className="css-0"
          >
            hispanic
          </th>
          <th
            className="css-0"
          >
            other_race
          </th>
          <th
            className="css-0"
          >
            male
          </th>
          <th
            className="css-0"
          >
            female
          </th>
          <th
            className="css-0"
          >
            unknown_sex
          </th>
          <th
            className="css-0"
          >
            %
          </th>
        </tr>
      </thead>
      <tbody
        className="css-0"
      >
        <tr
          className="css-0"
        >
          <td
            className="css-0"
          >
            798647
          </td>
          <td
            className="css-0"
          >
            116884
          </td>
          <td
            className="css-0"
          >
            119889
          </td>
          <td
            className="css-0"
          >
            1035420
          </td>
          <td
            className="css-0"
          >
            189630
          </td>
          <td
            className="css-0"
          >
            291084
          </td>
          <td
            className="css-0"
          >
            69485
          </td>
          <td
            className="css-0"
          >
            33868
          </td>
          <td
            className="css-0"
          >
            337264
          </td>
          <td
            className="css-0"
          >
            417251
          </td>
          <td
            className="css-0"
          >
            280905
          </td>
          <td
            className="css-0"
          >
            NaN
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;

exports[`<VoterPage/> <VoterPage/> first tag should be be <div/> and should match snapshot 1`] = `
<div
  className="css-0"
  id="VoterPage"
>
  <div
    className="css-9jay18"
    id="VoterPagePercentageSection"
  >
    <h2
      className="chakra-heading css-175t7kh"
    >
      Top Segment of All Voters: dem - 798647 - 77.13%
    </h2>
    <h2
      className="chakra-heading css-175t7kh"
    >
      Percentage of All Voters that are  - NaN - NaN%
    </h2>
    <div
      className="chakra-select__wrapper css-42jcko"
    >
      <select
        className="chakra-select css-1ik61og"
        id="VoterPageSegmentDropDown"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        value=""
      >
        <option
          value=""
        >
          Select Segment
        </option>
        <option
          value="dem"
        >
          dem
        </option>
        <option
          value="rep"
        >
          rep
        </option>
        <option
          value="other_party"
        >
          other_party
        </option>
        <option
          value="total"
        >
          total
        </option>
        <option
          value="white"
        >
          white
        </option>
        <option
          value="black"
        >
          black
        </option>
        <option
          value="hispanic"
        >
          hispanic
        </option>
        <option
          value="other_race"
        >
          other_race
        </option>
        <option
          value="male"
        >
          male
        </option>
        <option
          value="female"
        >
          female
        </option>
        <option
          value="unknown_sex"
        >
          unknown_sex
        </option>
      </select>
      <div
        className="chakra-select__icon-wrapper css-162mkon"
      >
        <svg
          aria-hidden={true}
          className="chakra-select__icon"
          focusable={false}
          role="presentation"
          style={
            {
              "color": "currentColor",
              "height": "1em",
              "width": "1em",
            }
          }
          viewBox="0 0 24 24"
        >
          <path
            d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
            fill="currentColor"
          />
        </svg>
      </div>
    </div>
  </div>
  <div
    className="chakra-table__container css-2l2gn2"
    id="VoterPageTable"
  >
    <table
      className="chakra-table css-0"
    >
      <caption
        className="css-18i7bby"
      >
        Philadelphia Qualified Voter Listing 2018
      </caption>
      <thead
        className="css-0"
      >
        <tr
          className="css-0"
        >
          <th
            className="css-0"
          >
            dem
          </th>
          <th
            className="css-0"
          >
            rep
          </th>
          <th
            className="css-0"
          >
            other_party
          </th>
          <th
            className="css-0"
          >
            total
          </th>
          <th
            className="css-0"
          >
            white
          </th>
          <th
            className="css-0"
          >
            black
          </th>
          <th
            className="css-0"
          >
            hispanic
          </th>
          <th
            className="css-0"
          >
            other_race
          </th>
          <th
            className="css-0"
          >
            male
          </th>
          <th
            className="css-0"
          >
            female
          </th>
          <th
            className="css-0"
          >
            unknown_sex
          </th>
          <th
            className="css-0"
          >
            %
          </th>
        </tr>
      </thead>
      <tbody
        className="css-0"
      >
        <tr
          className="css-0"
        >
          <td
            className="css-0"
          >
            798647
          </td>
          <td
            className="css-0"
          >
            116884
          </td>
          <td
            className="css-0"
          >
            119889
          </td>
          <td
            className="css-0"
          >
            1035420
          </td>
          <td
            className="css-0"
          >
            189630
          </td>
          <td
            className="css-0"
          >
            291084
          </td>
          <td
            className="css-0"
          >
            69485
          </td>
          <td
            className="css-0"
          >
            33868
          </td>
          <td
            className="css-0"
          >
            337264
          </td>
          <td
            className="css-0"
          >
            417251
          </td>
          <td
            className="css-0"
          >
            280905
          </td>
          <td
            className="css-0"
          >
            NaN
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
